### PR TITLE
Highlight feature results

### PR DESCRIPTION
**Pull request by puneet-pdx**
Related to issue: # https://devtopia.esri.com/runtime/kotlin/issues/4828

### Description:

We want to select the features returned as elements from a trace operation

### Summary of changes:

- Convert the elements received as the TraceElementResults to features
- Calculate the extent of the features to zoom to
- Cache the features after grouping them by layers, so they could be selected/unselected
- Conditionally cache the result extent. Cache features extent if not there then cache the geometry extent
- add logic to select/unselect features based on the result user selects

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/163/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt [hunk 1]
```diff
@@ -309,13 +322,24 @@ public class TraceState(
...
                     is UtilityElementTraceResult -> {
                         currentTraceElementResults = result.elements
+                        val featuresForElements = utilityNetwork.getFeaturesForElements(currentTraceElementResults).getOrThrow()
+                        withContext(Dispatchers.Default) {
+                            if (featuresForElements.isNotEmpty()) {
+                                currentTraceResultsExtent =
+                                    getResultFeaturesExtent(featuresForElements)
+                            }
+                            currentTraceFeatureResultsGroupByLayers =
+                                featuresForElements.groupBy { it.featureTable?.layer as FeatureLayer }
```
Comment (eri9000) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt:341]: i think this could be moved up to the if statement

**Review by eri9000 — APPROVED** [review]
LGTM

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Only high level comments:

- If I select a new trace, should all features de-select?

Tab selection issue repro:
- Have a couple trace results
- Tap new trace. Bottom menu correctly switches to trace configuration, but tab highlight at the top does not.
- Tap new trace again and switches to results.
- Tap new trace again and top tab and trace configuration correctly selected

**Comment by puneet-pdx** [comment]
@TADraeseke 
>If I select a new trace, should all features de-select? 

I assume you are going from results tab to the new trace tab? In that case no all features should not de-select as the features on the map that are selected are showing the result that is selected. Once a new trace is done then its results are selected. We don't unselect the results as soon as new trace is navigated to as the user might not want to do a new trace and would like to go back to the results screen. 

>Tab selection issue repro:

Have a couple trace results
Tap new trace. Bottom menu correctly switches to trace configuration, but tab highlight at the top does not.
Tap new trace again and switches to results.
Tap new trace again and top tab and trace configuration correctly selected 

This [issue](https://devtopia.esri.com/runtime/kotlin/issues/4790) was fixed with a separate [PR](https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/pull/622), this branch didn't have those changes. I have merged the feature branch to this one now. You should not see those issues.

**Review by TADraeseke — APPROVED** [review]
Nice! Looks good to me!

**Pull request merged**